v2/fthealth.go

```diff
@@ -0,0 +1,108 @@
+package v2
+
+import (
+	"sync"
+	"time"
+)
+
+type HealthCheck struct {
+	SystemCode  string
+	Name        string
+	Description string
+	Checks      []Check
+	Parallel    bool
+}
+
+type Check struct {
+	ID               string
+	Name             string
+	Severity         uint8
+	BusinessImpact   string
+	TechnicalSummary string
+	PanicGuide       string
+	Checker          func() (string, error)
+}
+
+func RunCheck(hc *HealthCheck) HealthResult {
+	return hc.health()
+}
+
+func (ch *HealthCheck) health() (result HealthResult) {
```

Comment:
My java background makes me think that the `setResultBasicInfo` method does something else (due tot the `set` prefix).
It makes me thing that the `HealthCheck` will have some fields set from the `HealthResult`, not the other way around... (which is visible once you analyze the method)

Can't we make `setResultBasicInfo` have the `HealthResult` as a receiver? This seems more natural to me.
Furthermore, the `setResultGlobalOK` could also be written this way. Then this method could look something like:
```
result.setBasicInfo(ch) // it sets some stuff in the restul from the ch, as you'd expect
ch.doChecks(ch.Parallel, &result)
result.setResultGlobalOK() // the global state of the result is set based on its internal values
```